unit_scaling/functional.py

```diff
@@ -186,6 +195,21 @@ def layer_norm(
     return F.layer_norm(input, normalized_shape, weight, bias, eps)
 
 
+@docstring_from(
+    torch.add,
+    short_description="Applies a **unit-scaled** addition.",
+    unsupported_args=["alpha"],
+)
+def add(
+    input: Tensor, other: Tensor, alpha: int = 1, out: Optional[Tensor] = None
+) -> Tensor:
+    input_broadcast_size, other_broadcast_size = _get_broadcast_sizes(input, other)
+    input = scale_bwd(input, input_broadcast_size**-0.5)
+    other = scale_bwd(other, other_broadcast_size**-0.5)
+    out = torch.add(input, other, out=out)
+    return scale_fwd(out, 2**-0.5)
```

Comment:
Maybe could just support `alpha`, just by forwarding it to `torch.add` & then `scale_fwd(out, (1+alpha)**-0.5)`? Or is this perhaps a bit risky / an anti-pattern?

Reply:
Yeah I thought a bit about whether I should push scaling factors through to alpha (and compensate accordingly) - in the end I decided it could be left for now and therefore would be left for now - if you think it should be added at this stage though please push back!

Reply:
No objection!